tests/test_segmentation.py

```diff
@@ -140,6 +141,35 @@ def test_find_label_issues():
         )
 
 
+@pytest.mark.slow
+def test_find_label_issues_memmap(tmp_path: Path):
+    """
+    Test that find_label_issues works with large memmap arrays
+    """
+    # for mem testing
+    start_time = time.time()
+    start_mem = max(memory_usage(interval=0.1))
+
+    # Create dummy versions of pred_probs and labels
+    # write to the pytest tmp_path so that the files are deleted after the test
+    pred_probs_file = tmp_path / "pred_probs.npy"
+    labels_file = tmp_path / "labels.npy"
+    np.save(pred_probs_file, np.random.rand(20, 5, 200, 200))
+    np.save(labels_file, np.random.randint(0, 2, (20, 200, 200)))
+
+    # Load the numpy arrays from disk and convert to zarr arrays
+    pred_probs = np.load(pred_probs_file, mmap_mode="r")
+    pred_labels = np.load(labels_file, mmap_mode="r")
+
+    _ = find_label_issues(pred_labels, pred_probs, n_jobs=None, batch_size=1000)
+
+    # for mem testing
+    end_mem = max(memory_usage(interval=0.1))
+    end_time = time.time()
+    print(f"Average memory used: {end_mem - start_mem} MiB")
+    print(f"Time taken: {end_time - start_time} seconds")
+
```

Comment:
can we add some sort of assert about the mem-usage?

Eg. this test could run the same dataset (that contains say 200 images) twice, one time with: batch_size = 200, one time with batch_size = 5. Verify mem-usage is much lower with batch_size = 5, and that the results match between the two different runs

Reply:
Great idea! You got it.

Reply:
Interesting...I'm trying to profile and compare these in the same run using 
```python
    # Test with low batch size
    peak_mem_low_batch_size = memory_usage(
        proc=(find_label_issues, (pred_labels, pred_probs), {'n_jobs': None, 'batch_size': 5}),
        interval=0.01,
        max_usage=True,
        include_children=True,
    )
    print(f"Peak memory used with batch size 5: {peak_mem_low_batch_size} MiB")

    # Test with high batch size
    peak_mem_high_batch_size = memory_usage(
        proc=(find_label_issues, (pred_labels, pred_probs), {'n_jobs': None, 'batch_size': 200}),
        interval=0.01,
        max_usage=True,
        include_children=True,
    )
    print(f"Peak memory used with batch size 200: {peak_mem_high_batch_size} MiB")
```
However, I notice whichever method gets called first uses higher memory. Naively searching leads to believe this is OS memory caching or Python garbage collection...but to eliminate those variables entirely and still effectively test mem I'll need to think of a better way to isolate them, so stay tuned...